12Oleg/Node.js_Practice
# HW1 v1

Objective
Create a Node.js application using the Express.js framework and integrate Swagger documentation. The application should also have a health-check endpoint and be hosted on GitHub. It should be run using Nodemon. 
Requirements
    1. Initialize a Node.js project:
        1. Set up a new Node.js project using npm or yarn.
        2. Install the necessary dependencies, including Express.js and Nodemon.
    2. Create an Express.js Application:
        1. Set up an Express.js application with a basic server.
        2. Create a route for a health-check endpoint (GET /health-check) that returns a JSON response indicating the server is running.
    3. Integrate Swagger Documentation:
        1. Install and set up Swagger UI and Swagger JS Docs for generating API documentation.
        2. Document at least one API endpoint (GET /health-check) 
    4. Host the Project on GitHub:
        1. Create a GitHub repository for your Node.js project.
        2. Push your project's source code and documentation to the GitHub repository.
    5. Test Your Application:
        1. Use Nodemon to run your application locally.
        2. Use a tool like Postman or CURL to test your health-check endpoint and ensure it returns the expected response.
        3. Access the Swagger documentation to verify that your API documentation is generated correctly.
    6. Additional Features (Optional, but encouraged):
        1. Add more endpoints to your Express.js application and document them using Swagger.
        2. Implement error handling for common HTTP status codes (e.g., 404 Not Found, 500 Internal Server Error) and document error responses in Swagger.
